src/core/ipfs/block.js

```diff
@@ -1,24 +1,47 @@
 'use strict'
 
+const Block = require('ipfs-block')
+const multihash = require('multihashes')
+
 module.exports = function block (self) {
   return {
-    get: (multihash, callback) => {
-      self._blockS.getBlock(multihash, callback)
+    get: (hash, callback) => {
+      if (typeof hash === 'string') {
+        hash = multihash.fromB58String(hash)
+      }
+      self._blockS.getBlock(hash, callback)
     },
     put: (block, callback) => {
-      self._blockS.addBlock(block, callback)
+      if (Array.isArray(block)) {
+        return callback(new Error('Array is not supported'))
+      }
+      if (Buffer.isBuffer(block)) {
+        block = new Block(block)
+      }
+
+      self._blockS.addBlock(block, (err) => {
+        callback(err, block)
+      })
     },
-    del: (multihash, callback) => {
-      self._blockS.deleteBlock(multihash, callback)
+    del: (hash, callback) => {
+      if (typeof hash === 'string') {
+        hash = multihash.fromB58String(hash)
+      }
+
+      self._blockS.deleteBlock(hash, callback)
     },
-    stat: (multihash, callback) => {
-      self._blockS.getBlock(multihash, (err, block) => {
+    stat: (hash, callback) => {
+      if (typeof hash === 'string') {
+        hash = multihash.fromB58String(hash)
+      }
```

Comment:
this is a lot of duplication, would be nice to simplify this to something like this

```js
const ensureHash = (hash) => {
  if (typeof hash === 'string') return multihash.fromB58String(hash)
  return hash
}
```